core/DataAccess/Model.php

```diff
@@ -75,12 +75,26 @@ public function getInvalidatedArchiveIdsSafeToDelete($archiveTable)
         $rows = Db::fetchAll($sql);
         foreach ($rows as $row) {
             $duplicateArchives = explode(',', $row['archives']);
-            $countOfArchives = count($duplicateArchives);
 
-            // if there is more than one archive, the older invalidated ones can be deleted
-            if ($countOfArchives > 1) {
-                array_shift($duplicateArchives); // we don't want to delete the latest archive if it is usable
+            // do not consider purging partial archives, if they are the latest archive,
+            // and we don't want to delete the latest archive if it is usable
+            while (!empty($duplicateArchives)) {
+                $pair = $duplicateArchives[0];
+                if (strpos($pair, '.') === false) {
+                    continue; // see below
+                }
+
+                list($idarchive, $value) = explode('.', $pair);
 
+                array_shift($duplicateArchives);
+
+                if ($value != ArchiveWriter::DONE_PARTIAL) {
```

Comment:
How would a done partial archive ever get invalidated I wonder? Eg say we invalidate the last 6 months and then rearchive all data. Would the partial archives still exist?

Reply:
It's always treated as invalidated, we never treat it like it's a full archive so if the latest is just a partial archive, we will initiate archiving when doing the "should we archive this" check.

> Eg say we invalidate the last 6 months and then rearchive all data. Would the partial archives still exist?

The partial archives would exist, but with a older ts_archived. We would not use the data in there, if there is more recent data we can use.

When purging, the partial archives that are older than a DONE_OK/DONE_INVALIDATED archive are removed.